File: jecnaapi-java/src/examples/java/ExamTopicsExample.java
import io.github.tomhula.jecnaapi.data.examTopics.ExamTopicCategory;
import io.github.tomhula.jecnaapi.data.examTopics.ExamTopicCategoryReference;
import io.github.tomhula.jecnaapi.data.examTopics.ExamTopicsPage;
import io.github.tomhula.jecnaapi.java.JecnaClientJavaWrapper;

public class ExamTopicsExample
{
    public static void main(String[] args)
    {
        /* Metody join() používáme aby jsme z asynchroního běhu "udělali" synchroní; pouze v příkladech */

        JecnaClientJavaWrapper client = new JecnaClientJavaWrapper();

        client.login("user", "password").join();

        // Get the main exam topics page with all categories
        ExamTopicsPage examTopicsPage = client.getExamTopicsPage().join();

        System.out.println("Available exam topic categories:");
        for (ExamTopicCategoryReference category : examTopicsPage.getCategories()) {
            System.out.println("- " + category.getName() + " (" + category.getUrl() + ")");
        }

        System.out.println();

        // Get topics for a specific category
        ExamTopicCategoryReference firstCategory = examTopicsPage.getCategories().stream().findFirst().orElse(null);
        if (firstCategory != null) {
            ExamTopicCategory categoryDetails = client.getExamTopicCategory(firstCategory).join();
            
            System.out.println("Topics for " + categoryDetails.getName() + ":");
            for (int i = 0; i < categoryDetails.getTopics().size(); i++) {
                System.out.println((i + 1) + ". " + categoryDetails.getTopics().get(i).getDescription());
            }
        }
    }
}

File: src/examples/kotlin/ExamTopicsExample.kt
import kotlinx.coroutines.runBlocking
import io.github.tomhula.jecnaapi.JecnaClient

fun main(): Unit = runBlocking {

    val client = JecnaClient()

    client.login("user", "password")

    // Get the main exam topics page with all categories
    val examTopicsPage = client.getExamTopicsPage()

    println("Available exam topic categories:")
    for (category in examTopicsPage.categories)
    {
        println("- ${category.name} (${category.url})")
    }

    println()

    // Get topics for a specific category
    val firstCategory = examTopicsPage.categories.firstOrNull()
    if (firstCategory != null)
    {
        val categoryDetails = client.getExamTopicCategory(firstCategory)

        println("Topics for ${categoryDetails.name}:")
        categoryDetails.topics.forEachIndexed { index, topic ->
            println("${index + 1}. ${topic.description}")
        }
    }
}

